python-checks/src/main/java/org/sonar/python/checks/IdentityComparisonWithCachedTypesCheck.java

```diff
@@ -56,21 +54,21 @@ public class IdentityComparisonWithCachedTypesCheck extends PythonSubscriptionCh
    * Fully qualified names of constructors and functions that would are guaranteed to create fresh objects with
    * references not shared anywhere else.
    * <p>
-   * If a reference arises from an call to one of these functions, and it does not escape anywhere else, then an
+   * If a reference arises from a call to one of these functions, and it does not escape anywhere else, then an
    * <code>is</code>-comparison with such a reference will always return <code>False</code>.
    * <p>
    * Note that these are fully qualified names of (value-level) expressions, not types.
    */
-  private static final Set<String> FQNS_CONSTRUCTORS_RETURNING_UNIQUE_REF = new HashSet<>(
-    asList("frozenset", "bytes", "int", "float", "str", "tuple", "hash"));
+  private static final List<String> FQNS_CONSTRUCTORS_RETURNING_UNIQUE_REF =
```

Comment:
Note: I'm confused to see this being further changed in this `propagate unary expressions` commit. I feel it would have been cleaner to squash this with the first commit.